gemini/use-cases/education/resume-generation/resume_generate.ipynb

```diff
@@ -0,0 +1,979 @@
+        "    Parses a docx file and converts it to text, handling tables and paragraphs.\n",
+        "\n",
+        "*   def parse_resume(res_path: str) -> str:\n",
+        "    Parses a resume from various file types (docx, pdf, txt) with fallbacks to handle missing personal information.\n",
+        "\n",
+        "*   def get_response(parsed_res: str, parsed_jd: str) -> str:\n",
+        "    Generates a resume using Gemini, incorporating extracted information, a professional summary, and arranged skills/coursework.\n",
+        "\n",
+        "*   def structure_response(resume_generated: str) -> str:\n",
+        "    Structures the generated resume into a specific format for readability."
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "metadata": {
+        "id": "c8dc8fe8-890c-4471-915a-396787e84a1d"
+      },
+      "outputs": [],
+      "source": [
+        "def check_info(resume: str) -> Dict[str, str]:\n",
+        "\n",
+        "    vertexai.init(\n",
+        "        project=\"aniket-tdcx\", location=\"us-central1\"\n",
+        "    )  # Initialize the AI Platform client with project ID\n",
+        "\n",
+        "    \"\"\"\n",
+        "    Check if the name, address, mobile number and email address is present in the resume\n",
+        "\n",
+        "    Args:\n",
+        "        resume: The parsed resume as str\n",
+        "\n",
+        "    Returns:\n",
+        "        Details requested in the prompt as dict\n",
+        "    \"\"\"\n",
+        "    prompt = f\"\"\"\n",
+        "    Extract the following candidate information from the given resume\n",
+        "    1. Name\n",
+        "    2. Address\n",
+        "    3. Mobile Number\n",
+        "    4. Email address\n",
+        "    Text : {resume}\n",
+        "\n",
+        "    INSTRUCTIONS:\n",
+        "    - In case the information is not present, return 'Not Found'\n",
+        "    - The final output should be in valid JSON format\n",
+        "    \"\"\"\n",
+        "    resp = model.generate_content(prompt).text\n",
+        "    return literal_eval(resp.split(\"```json\")[-1].split(\"```\")[0].replace(\"\\n\", \"\"))"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "metadata": {
+        "id": "e87930b0-a15b-4ad7-86fd-a941cebc1637"
+      },
+      "outputs": [],
+      "source": [
+        "def read_pdf(file_path: str) -> str:\n",
+        "    \"\"\"Extracts text from a PDF.\n",
+        "\n",
+        "    Args:\n",
+        "        file_path: path to the pdf file\n",
+        "\n",
+        "    Returns:\n",
+        "        resume content as plain text\n",
+        "    \"\"\"\n",
+        "    with open(file_path, \"rb\") as file:\n",
+        "        reader = pypdf.PdfReader(file)\n",
+        "        text = \"\"\n",
+        "        for page in reader.pages:\n",
+        "            text += page.extract_text()\n",
+        "    return text"
```

Comment:
You shouldn't need to do this. Gemini can read PDFs natively.

https://cloud.google.com/vertex-ai/generative-ai/docs/multimodal/document-understanding

Also, the sample resume is in DOCX. Should this example only use PDFs?

Reply:
Yes, agreed. In the earlier jumpstarts, we used Gemini multimodal capabilities to read PDFs in specific.

Please note that here the input file format is expected to be PDF, DOCX, or .txt. 
Therefore, there are OS constraints for converting DOCX files to images.

We have revised the code for PDF only till now. Thank you!